MetadataExtractor/StringValue.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Text;
+
+using JetBrains.Annotations;
+
+namespace MetadataExtractor
+{
+    public sealed class StringValue : IConvertible
+    {
+        private readonly byte[] _buffer;
+        private Encoding _encoder = Encoding.UTF8;
+
+        public StringValue([NotNull] byte[] bytes)
+        {
+            _buffer = bytes;
+        }
+
+        public StringValue([NotNull] byte[] bytes, Encoding defaultEncoder)
+        {
+            _buffer = bytes;
+            _encoder = defaultEncoder;
+        }
+
+        public byte[] Bytes
+        {
+            get { return _buffer; }
+        }
+
+        public void SetEncodingByName(string encodingName)
+        {
+            _encoder = Encoding.GetEncoding(encodingName);
+        }
+
+        public Encoding Encoder
+        {
+            get { return _encoder; }
+            set { _encoder = value; }
+        }
+
+        public int Length
```

Comment:
Probably should clarify this as `ByteCount` as `Length` may be misinterpreted as the number of characters in the string, which won't always be the same thing.